Sources/DistributedCluster/Plugins/ClusterSystem+Plugins.swift

```diff
@@ -114,3 +114,9 @@ internal struct AnyPluginKey: Hashable, CustomStringConvertible {
         }
     }
 }
+
+/// Way to hook into ClusterSystem's actor lifecycle events, specifically to `ready` and `resign`.
+public protocol PluginActorLifecycleHook {
```

Comment:
Should this extend `_Plugin`?

Reply:
Ah, so your idea is that it's still a plugin. Hm 🤔 I was more going into direction if you want to extend your current plugin, you can just conform it to additional protocol, like `class CustomPlugin: Plugin, ActorLifecyleHookable` or something. 
Though also making it directly a plugin also makes sense, you basically distinguish between normal and hookable plugins 🤔